config/initializers/gitlab_shell_secret_token.rb

```diff
@@ -14,6 +13,8 @@
   File.write(secret_file, token)
 end
 
-if File.exist?(Gitlab.config.gitlab_shell.path) && !File.exist?(gitlab_shell_symlink)
-  FileUtils.symlink(secret_file, gitlab_shell_symlink)
+# TODO May be removed after updating the default secret_file path...?
```

Comment:
Well, then you cannot assume that the user who run gitlab and gitlab-shell has read permission for the `.gitlab_shell_secret`… Symlink has the same permissions as the symlinked file.

You assume that gitlab and gitlab-shell are installed in `/home/git` all over the place, so why not just put `.gitlab_shell_secret` into `/home/git` as default? It’s simpler and more logical.